### PR TITLE
GetMailInfo method to assist automatic email creation

### PR DESCRIPTION
**Pull request by tabascq**
Note: This doesn't work via CORS despite many attempts to get that to work. Accordingly, it only works when something like CustomHtmlPageLoader is used.

Also includes a change to the FilesController to pass arguments along, which will be necessary should anyone figure out the CORS part.

### CODE REVIEW

**Comment by tabascq** [comment]
The many lines of changes to PuzzleApiController is due to a line endings issue. It doesn't look that extensive when whitespace changes are omitted. The only changes I made are to add the new GetMailInfoAsync method and the MailInfo class.

**Comment by morganbr** [comment]
Can you please fix your line endings? It's hard to review and messes up history

**Comment by tabascq** [comment]
I *did* fix line endings (because VS asked me to on opening this file). If you'd rather me leave them in a non-Windowsy state, I can do that. As noted above, the only things I actually changed here are to add the GetMailInfoAsync method and add the MailInfo class.

File: ServerCore/Pages/PuzzleApiController.cs [hunk 1]
```diff
@@ -1,65 +1,97 @@
...
+        [HttpGet]
+        [Route("api/puzzleapi/getmailinfo")]
+        public async Task<ActionResult<MailInfo>> GetMailInfoAsync(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            Team team = await UserEventHelper.GetTeamForCurrentPlayer(context, currentEvent, User, userManager);
+
+            MailInfo mailInfo = new MailInfo();
+
+            mailInfo.PuzzleName = await (from puzzle in context.Puzzles
```
Comment (morganbr) [inline ServerCore/Pages/PuzzleApiController.cs:61]: It would be good to check access to the puzzle

File: ServerCore/Pages/PuzzleApiController.cs [hunk 2]
```diff
@@ -1,65 +1,97 @@
...
+                                         where puzzle.ID == puzzleId
+                                         select puzzle.Name).FirstOrDefaultAsync();
+            mailInfo.TeamName = team.Name;
+
+            // replace commas with semicolons for better email support
+            mailInfo.TeamContactEmail = team.PrimaryContactEmail.Replace(',', ';');
```
Comment (morganbr) [inline ServerCore/Pages/PuzzleApiController.cs:67]: Does this need a null check? I thought this was optional

File: ServerCore/Pages/PuzzleApiController.cs [hunk 3]
```diff
@@ -1,65 +1,97 @@
...
+        /// Allows external puzzles to validate a team password
+        /// </summary>
+        /// <param name="teamPassword">Potential team password</param>
+        /// <returns>True if the password belongs to a team, false otherwise</returns>
+        [HttpGet]
+        [Route("api/puzzleapi/getmailinfo")]
+        public async Task<ActionResult<MailInfo>> GetMailInfoAsync(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            Team team = await UserEventHelper.GetTeamForCurrentPlayer(context, currentEvent, User, userManager);
```
Comment (morganbr) [inline ServerCore/Pages/PuzzleApiController.cs:57]: What happens if the player isn't on a team in these scenarios?
1. Not signed up for this event
2. Author
3. Admin

**Review by morganbr — APPROVED** [review]

File: ServerCore/Pages/PuzzleApiController.cs (hunk 1 above)
Comment (tabascq) [reply ServerCore/Pages/PuzzleApiController.cs]: Added some checks, thanks for the reminder!

File: ServerCore/Pages/PuzzleApiController.cs (hunk 2 above)
Comment (tabascq) [reply ServerCore/Pages/PuzzleApiController.cs]: Probably, just added one

File: ServerCore/Pages/PuzzleApiController.cs (hunk 3 above)
Comment (tabascq) [reply ServerCore/Pages/PuzzleApiController.cs:57]: Now verifies that you're on a team. No special handling of authors/admins, no need for such things.

**Pull request merged**